corehq/apps/es/management/commands/initialize_xforms_index_without_all.py

```diff
@@ -0,0 +1,91 @@
+import copy
+
+from django.core.management.base import BaseCommand
+from django.conf import settings
+
+from corehq.elastic import get_es_new
+from corehq.pillows.mappings.xform_mapping import XFORM_ES_TYPE, XFORM_MAPPING, XFORM_INDEX
+from corehq.pillows.mappings.reportxform_mapping import REPORT_XFORM_INDEX_INFO
+from corehq.pillows.mappings.user_mapping import USER_INDEX
+from pillowtop.es_utils import ElasticsearchIndexInfo, initialize_index_and_mapping
+from pillowtop.models import KafkaCheckpoint
+
+
+class Command(BaseCommand):
+    help = ("Adhoc command for ICDS xforms reindex")
+
+    def add_arguments(self, parser):
+        parser.add_argument('index_name')
+        parser.add_argument('number_of_shards', type=int)
+        parser.add_argument('number_of_replicas', type=int)
+        parser.add_argument(
+            '-c',
+            '--create-index',
+            action='store_true',
+            default=False,
+            help="Create and initialize index but don't reindex",
+        )
+        parser.add_argument(
+            '-r',
+            '--do-reindex',
+            action='store_true',
+            default=False,
+            help="Run reindex"
+        )
+
+    def handle(self, index_name, number_of_shards, number_of_replicas, **options):
+        create_index = options.get('create_index')
+        do_reindex = options.get('do_reindex')
+
+        if create_index:
+            mapping = copy.deepcopy(XFORM_MAPPING)
+            mapping["_all"] = {"enabled": False}
+
+            shard_settings = {
+                'number_of_shards': number_of_shards, 'number_of_replicas': number_of_replicas
+            }
+            if settings.ES_SETTINGS and 'xforms' in settings.ES_SETTINGS:
+                settings.ES_SETTINGS['xforms'].update(shard_settings)
+            elif settings.ES_SETTINGS:
+                settings.ES_SETTINGS['xforms'] = shard_settings
+            else:
+                settings.ES_SETTINGS = {'xforms': shard_settings}
```

Comment:
changing settings during runtime is a bit of an antipattern. Not sure the best option here but perhaps pass a ducktyped object to `initialize_index_and_mapping` instead of `ElasticsearchIndexInfo`?